### DISCUSSION TITLE
az cli doesn't use documented repo

### DISCUSSION THREAD

**Discussion by timja** [actions/virtual-environments#5822] [General questions] [answered]
https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-Readme.md#cli-tools

Says that it is installed from 'https://packages.microsoft.com/repos/azure-cli/'
It actually appears to come from somewhere else, likely https://packages.microsoft.com/ubuntu/18.04/prod

The stackoverflow post on downgrading your azure cli version no longer works because of that:
https://stackoverflow.com/a/61215035/4951015

It can't find the version:
```
E: Version '2.36.0-1~bionic' for 'azure-cli' was not found
```

Checking the available versions only reports 2.0.25-1
```
$ sudo apt-cache madison azure-cli

 azure-cli |   2.0.25-1 | https://packages.microsoft.com/ubuntu/18.04/prod bionic/main amd64 Packages
```

I was able to make it work by adding this to my pipeline:

```shell
AZ_REPO=$(lsb_release -cs)
echo "deb [arch=amd64] https://packages.microsoft.com/repos/azure-cli/ $AZ_REPO main" |
  sudo tee /etc/apt/sources.list.d/azure-cli.list

sudo apt-get update
sudo apt-cache madison azure-cli
sudo apt-get install -y --allow-downgrades azure-cli=2.36.0-1~bionic
```

I've had to do this because of breaking changes that do not have a replacement, (there is an issue raised already).

**Comment by al-cheb** [comment, marked as answer]
We install `azure-cli` from the  `repos/azure-cli` repo and delete it after the installation, that's why you should manually to add it at runtime - https://github.com/actions/virtual-environments/blob/main/images/linux/scripts/installers/azure-cli.sh

Also we provide the installation method in https://github.com/actions/virtual-environments/blob/main/images/linux/Ubuntu1804-Readme.md#cli-tools doc file which repo we use:
![image](https://user-images.githubusercontent.com/47745270/175928808-a1fe3880-4087-4fbd-920d-61588e05f786.png)

**Reply by timja** [reply]
Thanks! I've submitted an edit to the stackoverflow post (pending approval)
